jobs/email/ReviewReminder.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+/**
+ * @file jobs/email/ReviewReminder.php
+ *
+ * Copyright (c) 2014-2024 Simon Fraser University
+ * Copyright (c) 2000-2024 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class ReviewReminder
+ *
+ * @ingroup jobs
+ *
+ * @brief Class to handle a job to send an review reminder
+ */
+
+namespace PKP\jobs\email;
+
+use APP\facades\Repo;
+use APP\core\Services;
+use Illuminate\Support\Facades\Mail;
+use PKP\log\event\PKPSubmissionEventLogEntry;
+use PKP\core\PKPApplication;
+use PKP\core\Core;
+use PKP\invitation\invitations\ReviewerAccessInvite;
+use PKP\mail\mailables\ReviewResponseRemindAuto;
+use PKP\mail\mailables\ReviewRemindAuto;
+use PKP\jobs\BaseJob;
+
+
+class ReviewReminder extends BaseJob
+{
+    public function __construct(
+        public int $reviewAssignmentId,
+        public int $submissionId,
```

Comment:
The `$submissionId` and `$contextId` can be derived from the review assignment; would it be better not to serialize them then have to check for discrepancies? (For example, later on we look at the `$submissionId` to fetch the submission from the database -- but there's no check to ensure it matches the submission ID in the review assignment. It would be better just to use the `reviewAssignmentId`, then get the submission ID etc. from that.)

Reply:
I think thats better as we intend to keep the job serialized data as small as possible .

Reply:
However I think we need to set the standard to passing `contextId` whenever it's possible as our `Services` class bit too hard to mock.